app/src/main/java/ryey/easer/skills/usource/airplane_mode/AirplaneModeTracker.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2016 - 2022 Rui Zhao <renyuneyun@gmail.com>
+ *
+ * This file is part of Easer.
+ *
+ * Easer is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Easer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Easer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ryey.easer.skills.usource.airplane_mode;
+
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.provider.Settings;
+
+import androidx.annotation.NonNull;
+
+import ryey.easer.skills.condition.SkeletonTracker;
+
+public class AirplaneModeTracker extends SkeletonTracker<AirplaneModeUSourceData> {
+
+    private final AirplaneModeUSourceData condition;
+    private boolean curMode;
```

Comment:
Similar to above, the super constructor saves the current state in `satisfied` field (type `Boolean`). No need to repeat that.